### PR TITLE
initialise IamDataFrame with index

### LINKED ISSUE #77: IamDataFrame constructor should take pd.DataFrame with index
Currently, the `IamDataFrame.__init__()` can take a `pd.DataFrame` with columns `['model', 'scenario' , ...]`, but it fails if these identifiers are given as an index.

This would be a nice-to-have feature so that the `pd.DataFrame` returned by the `timeseries()` function can be turned back into an `IamDataFrame`. This currently requires a `reset_index()`.

### PR DESCRIPTION
**Pull request by danielhuppmann**
# Please confirm that this PR has done the following:

- [x] Tests Added
- [x] Documentation Added
- [x] Description in RELEASE_NOTES.md Added

# Description of PR

This PR extends the IamDataFrame constructor to take a pd.DataFrame with a meaningful index, for example the dataframe returned by `IamDataFrame.timeseries()`. Closes #77

### CODE REVIEW

**Comment by gidden** [comment]
is this good to go?

File: pyam/utils.py [hunk 1]
```diff
@@ -169,6 +169,10 @@ def format_data(df):
             df['scenario'] = scen.apply(
                 lambda s: '-'.join(s.split('-')[1:]).strip())
 
+    # reset the index if meaningful entries are included there
+    if not list(df.index.names) == [None]:
```
Comment (gidden) [inline pyam/utils.py]: here I think you could do just `if df.index.names:`

Comment (danielhuppmann) [reply pyam/utils.py]: just tried this, seems that this returns `True` even if no names are given

**Comment by danielhuppmann** [comment]
thanks @gidden, good to go after the tests pass!

**Comment by gidden** [comment]
Ah, of course, it has an element. Oh well.

**Pull request merged**